pkg/contracts/solana/inbound_test.go

```diff
@@ -67,6 +67,61 @@ func Test_ParseInboundAsDeposit(t *testing.T) {
 		// check result
 		require.EqualValues(t, expectedDeposit, deposit)
 	})
+
+	t.Run("should not parse if wrong discriminator", func(t *testing.T) {
+		txResult := LoadSolanaInboundTxResult(t, txHash)
+		tx, err := txResult.Transaction.GetTransaction()
+		require.NoError(t, err)
+
+		instruction := tx.Message.Instructions[0]
+
+		// try deserializing instruction as a 'deposit'
+		var inst DepositInstructionParams
+		err = borsh.Deserialize(&inst, instruction.Data)
+		require.NoError(t, err)
+
+		// serialize it back with wrong discriminator
+		data, err := borsh.Serialize(DepositInstructionParams{
+			Amount:        inst.Amount,
+			Discriminator: DiscriminatorDepositSPL,
+			Memo:          inst.Memo,
+		})
+		require.NoError(t, err)
+
+		tx.Message.Instructions[0].Data = data
+
+		deposit, err := ParseInboundAsDeposit(tx, 0, txResult.Slot)
+		require.NoError(t, err)
+		require.Nil(t, deposit)
```

Comment:
I don't this assertion is necessary. deposit should always be ignored if err is returned in any case